src/cloudWatchLogs/changeLogSearch.ts

```diff
@@ -53,39 +53,62 @@ export async function getNewData(
             }
             newData.parameters.startTime = isViewAllEvents(newTimeRange) ? undefined : newTimeRange.start
             newData.parameters.endTime = isViewAllEvents(newTimeRange) ? undefined : newTimeRange.end
-
             break
     }
+    let resourceType: telemetry.CloudWatchResourceType = 'logGroup'
+
     if (newData.logGroupInfo.streamName) {
         newData.retrieveLogsFunction = filterLogEventsFromUriComponents
         newData.parameters.streamNameOptions = [newData.logGroupInfo.streamName]
         newData.logGroupInfo.streamName = undefined
+        resourceType = 'logStream'
     }
 
+    recordTelemetryFilter(newData, resourceType, 'escapeHatch')
+
     return newData
 }
 
 export async function changeLogSearchParams(
     registry: LogStreamRegistry,
     param: 'filterPattern' | 'timeFilter'
 ): Promise<void> {
-    //let result: telemetry.Result = 'Succeeded'
     const oldUri = getActiveDocumentUri(registry)
 
     const oldData = registry.getLogData(oldUri)
     if (!oldData) {
+        telemetry.recordCloudwatchlogsFilter({
+            result: 'Failed',
+            source: 'Editor',
+            cloudWatchResourceType: isLogStreamUri(oldUri) ? 'logStream' : 'logGroup',
+            hasTimeFilter: param === 'timeFilter',
+            hasTextFilter: param === 'filterPattern',
+        })
         throw new Error(`cwl: Unable to find data for active URI ${oldUri}`)
     }
     const newData = await getNewData(param, oldData)
 
     if (!newData) {
-        //result = 'Cancelled'
+        telemetry.recordCloudwatchlogsFilter({
```

Comment:
If `recordCloudwatchlogsOpen` is for "opening a log stream without filtering" then we shouldn't record it for the filter case.

Alternatively, we don't need `recordCloudwatchlogsFilter` at all. We could add filter/search meatadata to `recordCloudwatchlogsOpen`. I don't see why we need both.

Reply:
I believe VS uses "open" to refer to the editor being opened up while "filter" refers to any user gestures that change filters on the editor. 

This distinction is less relevant for us and can be captured via "source". I wouldn't try to overload `recordCloudwatchlogsOpen` too much though. I'd suggest using `recordCloudwatchlogsFilter` for when the editor is already opened and `recordCloudwatchlogsOpen` otherwise.